### PR TITLE
Make addresses in To header unique when group replying 

### PR DESCRIPTION
**Pull request by pazz**
This is just relaying the patch proposed in #818. related: #874.

`make_unique` should at least be documented properly, maybe made into a classmethod.
 The docstring of `clear_my_address` need to be changed.

### CODE REVIEW

File: alot/commands/thread.py [hunk 1]
```diff
@@ -221,15 +221,27 @@ def apply(self, ui):
...
                         my_addresses, mail.get_all('Cc', []))
-                    envelope.add('Cc', decode_header(cc))
+                    envelope.add('Cc', decode_header(', '.join(cc)))
+
+        def make_unique(value):
+            res = dict()
+            for name, address in getaddresses(value):
+                res[address] = name
+            new_value = []
+            for address, name in res.items():
```
Comment (dcbaker) [inline alot/commands/thread.py]: you should use `res.iteritems()` instead of `res.items()`

File: alot/commands/thread.py [hunk 2]
```diff
@@ -221,15 +221,27 @@ def apply(self, ui):
...
+                if name != '':
+                    new_value.append('"%s" <%s>' % (name, address))
+                else:
+                    new_value.append(address)
```
Comment (dcbaker) [inline alot/commands/thread.py]: What about using a comprehension?
```python
sorted(['"{}" <{}>'.format(n, a) if n != '' else a for n, a in res.iteritems()])
```
Or do you think having the ternary in the comprehension is too complex?

Comment (pazz) [inline alot/commands/thread.py]: i wonder  why he used `sorted` here anyway..

File: alot/commands/thread.py (hunk 1 above)
Comment (pazz) [inline alot/commands/thread.py]: ok

File: alot/commands/thread.py (hunk 2 above)
Comment (pazz) [inline alot/commands/thread.py]: i used your comprehension now, please check if the formatting makes sense.

**Comment by pazz** [comment]
oh wait! something is wrong! the current branch here mixes up address and names for a subset of the recipients when group replying!

**Comment by pazz** [comment]
ah ok, (iter)items() returns pairs key,value not the other way around.
I'm happy with this now.

**Comment by pazz** [comment]
the docs tests seem to fail here, and i cannot see why. This should be totally unrelated. Bug in the travis setup?

```
Warning, treated as error:

/home/travis/build/pazz/alot/docs/source/usage/commands.rst:27: WARNING: usage/modes/global is already assigned section numbers (nested numbered toctree?)
```

**Comment by lucc** [comment]
I can not reproduce this locally. You can try to change something in line 102 of the travis file. One thing could be to `make cleanall html` instead of just `html` and you could increase the verbosity of sphinx-build with one or more `-v` options.

**Comment by pazz** [comment]
I also cannot reproduce this locally. the verbose debug message does not really tell me anything as well.
I've split the commit into two smaller ones, but each one makes travis fail the docs build.
This looks completely unrelated to me!

**Comment by pazz** [comment]
@lucc: the craziest thing is that the docs also fail on a new test branch (off master, same HEAD), whereas not on master itself.

**Comment by lucc** [comment]
Maybe there is something broken in the cache? Can you clear the cache?

**Comment by dcbaker** [comment]
I deleted the cache on one of the branches and restarted it. We'll see if that fixes the issue.

**Comment by dcbaker** [comment]
There's a new version of sphinx. I wonder if that's the problem? 1.5.2 vs 1.5.1

**Comment by dcbaker** [comment]
To be clear, clearing the cache did not solve the problem

**Comment by lucc** [comment]
I had a look at the actual sphinx warning 
```
SphinxWarning: /home/travis/build/pazz/alot/docs/source/usage/commands.rst:27: WARNING: usage/modes/global is already assigned section numbers (nested numbered toctree?)
```
and the lines 27+ in this file are not generating any html output as far as I can see. So I removed them but then I get another warning:
```
SphinxWarning: /home/luc/vcs/alot/docs/source/usage/modes/bufferlist.rst:: WARNING: document isn't included in any toctree
```
Maybe there is something wrong with this part of the file. But I don't know rst or sphinx so I hope you have an idea how to fix that.

**Comment by pazz** [comment]
OK, we can treat the travis/docs issue separately: I just triggered a rebuild on the master branch and get the same problem. I believe this was indeed related to the cache somehow hiding the fact that an earlier commit already resulted in a fail. #994

**Comment by pazz** [comment]
@dcbaker I addressed your previous review, can you have another look and OK/merge this one then?

**Comment by dcbaker** [comment]
lgtm

**Comment by pazz** [comment]
thanks

**Pull request merged**